nav2_smac_planner/CMakeLists.txt

```diff
@@ -1,153 +1,117 @@
+)
 
 include_directories(
-  include
-  ${OMPL_INCLUDE_DIRS}
+        include
+        ${catkin_INCLUDE_DIRS}
+        ${OMPL_INCLUDE_DIRS}
 )
 
-set(library_name nav2_smac_planner)
-
-set(dependencies
-  rclcpp
-  rclcpp_action
-  rclcpp_lifecycle
-  std_msgs
-  visualization_msgs
-  nav2_util
-  nav2_msgs
-  nav_msgs
-  geometry_msgs
-  builtin_interfaces
-  tf2_ros
-  nav2_costmap_2d
-  nav2_core
-  pluginlib
-  angles
-  eigen3_cmake_module
-)
+set(library_name ${PROJECT_NAME})
 
 # Hybrid plugin
 add_library(${library_name} SHARED
-  src/smac_planner_hybrid.cpp
-  src/a_star.cpp
-  src/collision_checker.cpp
-  src/smoother.cpp
-  src/analytic_expansion.cpp
-  src/node_hybrid.cpp
-  src/node_lattice.cpp
-  src/costmap_downsampler.cpp
-  src/node_2d.cpp
-  src/node_basic.cpp
+        src/smac_planner_hybrid.cpp
+        src/a_star.cpp
+        src/collision_checker.cpp
+        src/smoother.cpp
+        src/analytic_expansion.cpp
+        src/node_hybrid.cpp
+        src/node_lattice.cpp
+        src/costmap_downsampler.cpp
+        src/node_2d.cpp
+        src/node_basic.cpp
 )
 
-target_link_libraries(${library_name} ${OMPL_LIBRARIES})
-target_include_directories(${library_name} PUBLIC ${Eigen3_INCLUDE_DIRS})
-
-ament_target_dependencies(${library_name}
-  ${dependencies}
+target_link_libraries(${library_name}
+        ${catkin_LIBRARIES}
+        ${OMPL_LIBRARIES}
 )
 
-# 2D plugin
-add_library(${library_name}_2d SHARED
-  src/smac_planner_2d.cpp
-  src/a_star.cpp
-  src/smoother.cpp
-  src/collision_checker.cpp
-  src/analytic_expansion.cpp
-  src/node_hybrid.cpp
-  src/node_lattice.cpp
-  src/costmap_downsampler.cpp
-  src/node_2d.cpp
-  src/node_basic.cpp
-)
-
-target_link_libraries(${library_name}_2d ${OMPL_LIBRARIES})
-target_include_directories(${library_name}_2d PUBLIC ${Eigen3_INCLUDE_DIRS})
-
-ament_target_dependencies(${library_name}_2d
-  ${dependencies}
-)
-
-# Lattice plugin
-add_library(${library_name}_lattice SHARED
-  src/smac_planner_lattice.cpp
-  src/a_star.cpp
-  src/smoother.cpp
-  src/collision_checker.cpp
-  src/analytic_expansion.cpp
-  src/node_hybrid.cpp
-  src/node_lattice.cpp
-  src/costmap_downsampler.cpp
-  src/node_2d.cpp
-  src/node_basic.cpp
-)
-
-target_link_libraries(${library_name}_lattice ${OMPL_LIBRARIES})
-target_include_directories(${library_name}_lattice PUBLIC ${Eigen3_INCLUDE_DIRS})
-
-ament_target_dependencies(${library_name}_lattice
-  ${dependencies}
-)
-
-pluginlib_export_plugin_description_file(nav2_core smac_plugin_hybrid.xml)
-pluginlib_export_plugin_description_file(nav2_core smac_plugin_2d.xml)
-pluginlib_export_plugin_description_file(nav2_core smac_plugin_lattice.xml)
-
-install(TARGETS ${library_name} ${library_name}_2d ${library_name}_lattice
-  ARCHIVE DESTINATION lib
-  LIBRARY DESTINATION lib
-  RUNTIME DESTINATION lib/${PROJECT_NAME}
+# TODO 2D plugin
```

Comment:
I keep the code for the other 2 plugins (lattice and 2d) because they are partially ported through find-replaces, so it should be very fast to put on working order